inference-engine/src/mkldnn_plugin/nodes/base.hpp

```diff
@@ -54,6 +56,61 @@ class ExtLayerBase: public ILayerExecImpl {
     }
 
 protected:
+    class DataConfigurator {
+    public:
+        explicit DataConfigurator(MKLDNNPlugin::TensorDescCreatorTypes tensorDescType):
+                tensorDescType(tensorDescType) {}
+
+        DataConfigurator(MKLDNNPlugin::TensorDescCreatorTypes tensorDescType, Precision prc):
+                tensorDescType(tensorDescType), prc(prc) {}
+
+        DataConfigurator(MKLDNNPlugin::TensorDescCreatorTypes tensorDescType, Precision prc, bool constant, int inplace = -1):
+                tensorDescType(tensorDescType), prc(prc), constant(constant), inplace(inplace) {}
+
+        const MKLDNNPlugin::TensorDescCreatorTypes tensorDescType;
+        const bool constant = false;
+        const int inplace = -1;
+        const Precision prc = Precision::UNSPECIFIED; // By default ngraph node precision is used
+    };
+
+    void addConfig(const std::shared_ptr<ngraph::Node>& op,
+                   const std::vector<DataConfigurator>& inDataConfigurators,
+                   const std::vector<DataConfigurator>& outDataConfigurators,
+                   bool dynBatchSupport = false) {
+        LayerConfig config;
+
+        if (inDataConfigurators.size() != op->get_input_size())
+            THROW_IE_EXCEPTION << "Cannot add config for operation " << op->get_friendly_name() << ". Incorrect number of inputs: " <<
+                                  "expected: " << op->get_input_size() << ", provided: " << inDataConfigurators.size();
+        if (outDataConfigurators.size() != op->get_output_size())
+            THROW_IE_EXCEPTION << "Cannot add config for operation " << op->get_friendly_name() << ". Incorrect number of outputs: " <<
+                               "expected: " << op->get_output_size() << ", provided: " << outDataConfigurators.size();
+
+        auto fill_port = [] (const DataConfigurator& dataConfigurator, const ngraph::descriptor::Tensor& tensor, std::vector<DataConfig>& port) {
+            auto& creators = MKLDNNPlugin::TensorDescCreator::getCommonCreators();
+            if (creators.find(dataConfigurator.tensorDescType) == creators.end()) {
+                THROW_IE_EXCEPTION << "Cannot find tensor descriptor creator";
+            }
+            auto precision = dataConfigurator.prc != Precision::UNSPECIFIED ? dataConfigurator.prc : details::convertPrecision(tensor.get_element_type());
+
+            DataConfig dataConfig;
+            dataConfig.inPlace = dataConfigurator.inplace;
+            dataConfig.constant = dataConfigurator.constant;
+            dataConfig.desc = creators.at(dataConfigurator.tensorDescType)->createDesc(precision, tensor.get_shape());
```

Comment:
We need to define the behavior when the tensor rank is less then the allowed in the creator. Whether we just skip such call or raise an exception?  We still need to add such a check, since we are not using filters (e.g. `makeFilteredRange` call).

Reply:
Totally agree. I chose variant with skipping such call in order not to overcomplicated particular node initialization logic.